traversal/procedure/ProcedureEdge.java

```diff
@@ -1157,25 +1195,23 @@ public Forwardable<KeyValue<ThingVertex, ThingVertex>, Order.Asc> branchEdge(Gra
                             ).filter(kv -> kv.key().equals(player));
                             iter.forward(KeyValue.of(player, null));
                         } else {
-                            iter = instanceRoleTypes.mergeMap(
-                                    rt -> iterate(to.props().types())
-                                            .map(l -> graphMgr.schema().getType(l))
-                                            .mergeMap(
-                                                    t -> rel.outs()
-                                                            .edge(ROLEPLAYER, rt, PrefixIID.of(t.encoding().instance()), t.iid())
-                                                            .toAndOptimised(),
-                                                    ASC
-                                            ),
-                                    ASC
-                            );
+                            iter = instanceRoleTypes.flatMap(
+                                    rt -> {
+                                        return iterate(to.props().types())
+                                                .map(l -> graphMgr.schema().getType(l))
+                                                .map(t -> rel.outs()
+                                                        .edge(ROLEPLAYER, rt, PrefixIID.of(t.encoding().instance()), t.iid())
+                                                        .toAndOptimised()
+                                                );
+                                    }).mergeMap(Function.identity(), ASC);
```

Comment:
we remove 1 `mergeMap` call that could incur some overheads in terms of creating extra priority queues under the hood